tests/dialects/test_memref.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+
+from xdsl.ir import OpResult, SSAValue
+from xdsl.dialects.builtin import i32, IntegerType, IndexType
+from xdsl.dialects.memref import Alloc, Alloca, Dealloc, Dealloca, MemRefType, Load, Store
+
+
+def test_memreftype():
+    mem1 = MemRefType.from_type_and_list(i32)
+    assert mem1.get_num_dims() == 1
+    assert mem1.get_shape() == [1]
+
+    mem2 = MemRefType.from_type_and_list(i32, [3, 3, 3])
+    assert mem1.get_num_dims() == 1
+    assert mem2.get_shape() == [3, 3, 3]
+
+    my_i32 = IntegerType.from_width(32)
+    mem3 = MemRefType.from_params(my_i32)
+    assert mem3.get_num_dims() == 1
+    assert mem3.get_shape() == [1]
+
+
+def test_memref_load_i32():
+    i32_memref_type = MemRefType.from_type_and_list(i32)
+    memref_ssa_value = SSAValue(i32_memref_type)
+    load = Load.get(memref_ssa_value, [])
+
+    assert load.memref is memref_ssa_value
```

Comment:
Not sure whether to always use is here or ==. What do you think @math-fehr?

Currently, this is the case, but it might be something we might want to change at some point?

Reply:
`is` is more robust for this object, I think.
We should be using is for objects anytime

Reply:
I was thinking whether we'd change the implementation at some point that instead of copying the references to objects into our core datastructures, we (shallow or deep) copy the objects themselves. Not sure that makes sense though, as this would make the API quite weird...

Reply:
I don't think we want in the near future to start copying things, and I think it is better to assume that they are equal with `is` here. So I think it is fine to test it with `is`, rather than using `==`. In any way, if we change it, we can change the test.